doc/generate_module_rst.py

```diff
@@ -186,14 +188,19 @@ def prune_duplicates(self) -> None:
             new_classes = []
             for c in module.classes:
                 if id(c.obj) not in seen:
-                    seen.add(id(c.obj))
+                    # Special treatment for aliases, i.e. when names don't match.
+                    # Don't mark them as seen to avoid the original object being skipped.
+                    if c.name.endswith("." + c.obj.__name__):
+                        seen.add(id(c.obj))
                     new_classes.append(c)
             module.classes = new_classes
 
             new_functions = []
             for f in module.functions:
                 if id(f.obj) not in seen:
-                    seen.add(id(f.obj))
+                    # See comment above (for classes) about aliases.
```

Comment:
Thinking about this a bit more, wouldn't it make more sense to just use a different key? That way we would also remove duplicate aliases.

```suggestion
                    key = (id(f.obj), f.name[f.name.rfind(".")+1:])
                    if key not in seen:
                        seen.add(key)
                        new_functions.append(f)
```

Reply:
That makes sense. I have updated the keys.
This required changes to the dispatcher part as well to use the function implementation `__name__` in the key, so that it then later matches correctly here.

Reply:
This addresses your previous comment as well I think. The check is now against the implementation instead of the dispatcher itself.